kernel/kernel-impl/src/main/java/org/sakaiproject/authz/impl/DbAuthzGroupService.java

```diff
@@ -685,10 +685,7 @@ public void run() {
 					long time = 0;
 					long start = System.currentTimeMillis();
 					try {
-						// only remove from the cache if the realm was updated during the refresh
-						if (((DbStorage) m_storage).refreshAuthzGroupInternal((BaseAuthzGroup) azGroup)) {
-							m_realmRoleGRCache.remove(azGroupId);
```

Comment:
Nope as that only occurred in a thread processing the queue, it now fires an event which will then do the invalidation.